runtime/parachains/src/paras.rs

```diff
@@ -559,7 +597,14 @@ impl<T: Config> Module<T> {
 				for (para_id, _) in pruning_tasks_to_do {
 					let full_deactivate = <Self as Store>::PastCodeMeta::mutate(&para_id, |meta| {
 						for pruned_repl_at in meta.prune_up_to(pruning_height) {
-							<Self as Store>::PastCode::remove(&(para_id, pruned_repl_at));
+							let removed_code_hash =
+								<Self as Store>::PastCodeHash::take(&(para_id, pruned_repl_at));
+
+							if let Some(removed_code_hash) = removed_code_hash {
+								Self::decrease_code_ref(&removed_code_hash);
+							} else {
+								todo!("log error");
```

Comment:
I guess we can't do much other than log error?

Reply:
Yeah, we can't. We need to add that log